### PR TITLE
Add release pipeline

### PR DESCRIPTION
**Pull request by cynicaljoy**
Ticket(s): BT-3624

Adding a Concourse pipeline to automate the release process. Process still requires somebody we manually increment the version and manually trigger the run of the release job.

**Important:** The v4 and v5 pipelines use an API token the restricts access to `faunadb` and since we're publishing to a new project `fauna` I've had to provision an API key that has "Full Access" so that we can publish the new project. After the initial release we'll want to provision a new API Key that is restricted to just `fauna` and update the secrets accordingly.

### CODE REVIEW

File: concourse/scripts/integration-tests.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+set -eou
+
+apk add --update make
+
+#pip install .
+#pip install nose2
+#pip install pylint
+#pip install requests
+#make test
```
Comment (pnwpedro) [inline concourse/scripts/integration-tests.sh:11]: let's delete these if they aren't being used

File: concourse/scripts/publish.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+PACKAGE_VERSION=$(python setup.py --version)
+echo "Going to publish python package: ${PACKAGE_VERSION}"
+
+pip install twine
+
+twine check dist/*
+twine upload dist/*
+
+echo "fauna-python@$PACKAGE_VERSION has been released" > ../slack-message/publish
```
Comment (pnwpedro) [inline concourse/scripts/publish.sh:17]: I know you'll care, so adding this. ;)
```suggestion
echo "fauna-python@$PACKAGE_VERSION has been released" > ../slack-message/publish

```

File: concourse/scripts/publish_docs.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+sed -i.bak "0,/<\/title>/{s/<\/title>/<\/title>${HEAD_GTM}/}" ./docs/index.html
+
+echo "================================="
+echo "Adding google manager tag to body"
+echo "================================="
+
+BODY_GTM=$(cat ./fauna-python-repository/concourse/scripts/body_gtm.dat)
+sed -i.bak "0,/<body>/{s/<body>/<body>${BODY_GTM}/}" ./docs/index.html
+
+rm ./docs/index.html.bak
```
Comment (pnwpedro) [inline concourse/scripts/publish_docs.sh:38]: Can you add a little more comment detail to this section?

e.g.
* Why we are making a `.bak` and then deleting it
* Why the tag goes after title but inside body (maybe just a link to docs about this)

File: concourse/tasks/integration.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+version: "3.3"
```
Comment (pnwpedro) [inline concourse/tasks/integration.yml:1]: this file is in concourse/tasks/, which makes it seem like it should be a task, but it's a docker compose file? let's call it `docker-compose-fauna.yml`

File: concourse/tasks/integration.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    container_name: mytests-311
+    depends_on:
+      - faunadb
+    volumes:
+      - "../../:/tmp/app"
+    working_dir: "/tmp/app"
+    command:
+      - concourse/scripts/integration-tests.sh
+
+  tests-310:
```
Comment (pnwpedro) [inline concourse/tasks/integration.yml:31]:
```suggestion
  python-310:
```

File: concourse/tasks/integration.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      retries: 30
+
+  tests-311:
+    environment:
+      FAUNA_ROOT_KEY: "secret"
+      FAUNA_DOMAIN: faunadb
+      FAUNA_SCHEME: "http"
+      FAUNA_PORT: "8443"
+    image: python:3.11.2-alpine3.17
+    container_name: mytests-311
```
Comment (pnwpedro) [inline concourse/tasks/integration.yml:22]:
```suggestion
    container_name: python-311
```

File: concourse/tasks/integration.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      - concourse/scripts/integration-tests.sh
+
+  tests-310:
+    environment:
+      FAUNA_ROOT_KEY: "secret"
+      FAUNA_DOMAIN: faunadb
+      FAUNA_SCHEME: "http"
+      FAUNA_PORT: "8443"
+    image: python:3.10.10-alpine3.17
+    container_name: mytests-310
```
Comment (pnwpedro) [inline concourse/tasks/integration.yml:38]:
```suggestion
    container_name: python-310
```

File: concourse/scripts/publish_docs.sh (hunk 3 above)
Comment (cynicaljoy) [reply concourse/scripts/publish_docs.sh:38]: this is a copy and paste job 😁 -- https://github.com/fauna/faunadb-python/blob/v4/concourse/scripts/publish_docs.sh

**Pull request merged**